zokrates_core/src/typed_absy/folder.rs

```diff
@@ -711,6 +719,19 @@ pub fn fold_struct_expression<'ast, T: Field, F: Folder<'ast, T>>(
     }
 }
 
+pub fn fold_constant_symbol<'ast, T: Field, F: Folder<'ast, T>>(
+    f: &mut F,
+    p: TypedConstantSymbol<'ast, T>,
+) -> TypedConstantSymbol<'ast, T> {
+    match p {
+        TypedConstantSymbol::Here(tc) => TypedConstantSymbol::Here(TypedConstant {
```

Comment:
We should define `fold_constant` on `TypedConstant` in the trait and use that here

Reply:
Same in result_folder